### PR TITLE
MDTZ-978: Validate Figma API response schema

### PR DESCRIPTION
**Pull request by akostevich-atlassian**
## Changes

- **feat:** Validates Figma API response schema
- **refactor:** Use endpoint paths for schema IDs
- **refactor:** Fix multiple issues and inconsistencies in tests and clients identified thanks to added schemas.

### CODE REVIEW

File: src/infrastructure/jira/schemas.ts [hunk 1]
```diff
@@ -16,3 +23,41 @@ export const ATTACHED_DESIGN_URL_V2_VALUE_SCHEMA: JSONSchemaTypeWithId<
 		required: ['url', 'name'],
 	},
 };
+
+export const NODE_DEV_STATUS_SCHEMA: JSONSchemaType<NodeDevStatus> = {
```
Comment (akostevich-atlassian) [inline src/infrastructure/jira/schemas.ts:27]: **nitpick:** @stephbrook11 , consider exporting only top-level schemas (e.g., `FILE_RESPONSE_SCHEMA`) and keeping other schemas private, as for example, `NODE_DEV_STATUS_SCHEMA`.

Comment (akostevich-atlassian) [inline src/infrastructure/jira/schemas.ts:27]: **issue:** @stephbrook11 , please, move newly added schemas to `src/infrastructure/figma/figma-client/schemas.ts` as they belong to the Figma API contract.

File: src/infrastructure/figma/figma-service.ts [hunk 2]
```diff
@@ -62,6 +64,9 @@ export class FigmaService {
 				},
 				accessToken,
 			);
+
+			parseJsonOfSchemaFromObject(fileResponse, FILE_RESPONSE_SCHEMA);
```
Comment (akostevich-atlassian) [inline src/infrastructure/figma/figma-service.ts:68]: **issue:** Please, move the schema validation to `FigmaClient.getFile`:
https://github.com/atlassian-labs/figma-for-jira/blob/1e03d649818d8b57bedb34287afe9ddfa57155d8/src/infrastructure/figma/figma-client/figma-client.ts#L132

It better to encapsulate this responsibility in `FigmaClient` for the following reasons:

- `FigmaClient` guarantees that its methods return data of the specified type, so its consumers should not worry about this concern.
- Schema validation is encapsulated in a single place but not duplicated among consumers.

Please, see the `JiraClient` implementation for an example:
https://github.com/atlassian-labs/figma-for-jira/blob/1e03d649818d8b57bedb34287afe9ddfa57155d8/src/infrastructure/jira/jira-client/jira-client.ts#L110

File: src/infrastructure/ajv.ts [hunk 3]
```diff
@@ -53,6 +53,24 @@ export function parseJsonOfSchema<T>(
 	}
 }
 
+export function parseJsonOfSchemaFromObject<T>(
```
Comment (akostevich-atlassian) [inline src/infrastructure/ajv.ts:56]: **issue:** This method can be unnecessary -- consider using the existing `assertSchema` instead.

File: src/usecases/connect-figma-team-use-case.test.ts [hunk 4]
```diff
@@ -36,9 +36,9 @@ describe('connectFigmaTeamUseCase', () => {
 		);
 
 		// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-		const generatedPasscode = (
-			figmaService.createFileUpdateWebhook as jest.Mock
-		).mock.calls[0][1];
+		const generatedPasscode =
+			// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
```
Comment (sbbrook) [inline src/usecases/connect-figma-team-use-case.test.ts:40]: @akostevich-atlassian it's failing the pre-commit checks without this line

**Review by sbbrook — APPROVED** [review]

**Pull request merged**